### ISSUE TITLE
Support for multiple certs for same IdP?

### ISSUE THREAD

**Issue by ssypi** [bergie/passport-saml#189]
Any way to have IdP with multiple certs for verifying responses from the IdP? We need to be able to support two IdP certificates during a rollover/transition period when an IdP wishes to change their certificate.

**Comment by pdspicer** [comment]
Currently no support for multiple certs, would welcome a PR for this though!

**Comment by cjbarth** [comment]
Can the `pr-welcome` tag be added to this then?

**Comment by cjbarth** [comment]
What method would we like to use to configure such a thing? We need to determine how the configuration will work and where we might actually do the check and how. Would we have a primary/secondary cert and fail-over, or would we have a cert ID and another setting that dictates which cert to use before trying others?

**Comment by cjbarth** [comment]
BTW, the documentation states that you can already support multiple IdP certs for when they roll their certificate.

```
If the Identity Provider has multiple signing certificates that are valid (such as during the rolling from an old key to a new key and responses signed with either key are valid) then the cert configuration key can be an array:

    cert: [ 'MIICizCCAfQCCQCY8tKaMc0BMjANBgkqh ... W==', 'MIIEOTCCAyGgAwIBAgIJAKZgJdKdCdL6M ... g=' ]
```

What we really need is something similar for when the SP needs to roll it's certificate so that the SP doesn't go offline during re-federation with the IdP

**Comment by markstos** [comment]
According to @cjbarth, this sounds like it was resolved by #218.

If you want to open a new issue about multiple certs for the SP instead of the IdP, open a new issue for that (although no else is likely to implement it on your behalf, so submit a PR instead).
